src/gobprepare/prepare_client.py

```diff
@@ -254,6 +261,72 @@ def action_check_row_counts(self, action: RowCountConfig) -> bool:
                 )
         return tables_counts_ok
 
+    def action_check_source_sync_complete(self, action: syncSchemaConfig) -> bool:
+        """Check import action. Verify when sync tables from databricks is completed.
+
+        :param action:
+        :return:
+        """
+
+        public_schema = action["public_schema"]
+        table_name = action["table_name"]
+        schema = action["schema"]
+        if not self._dst_datastore.query(check_table_existence_query(public_schema, table_name)):
+            raise GOBException(f"Table '{public_schema}.{table_name}' does not exists.")
+        else:
+            try:
+                where_conditions = [{ "sync_schema": schema }]
+                select_query = create_select_where_query(table_name, where_conditions)
+                record = next(self._dst_datastore.query(select_query))
+            except StopIteration:
+                raise GOBException(f"No record for schema '{schema}' found in '{public_schema}.{table_name}'.")
+            else:
+                self._execute_update_sync_schema(schema, table_name, record, update_action = 'start_prepare')
+
+    def action_complete_prepare(self, action: syncSchemaConfig) -> None:
+        """update sync table after prepare is completed.
+
+        :param action:
+        :return:
+        """
+        public_schema = action["public_schema"]
+        table_name = action["table_name"]
+        schema = action["schema"]
+        try:
+            where_conditions = [{ "sync_schema": schema }]
+            select_query = create_select_where_query(table_name, where_conditions)
+            record = next(self._dst_datastore.query(select_query))
+        except StopIteration:
+            raise GOBException(f"No record for schema '{schema}' found in '{public_schema}.{table_name}'.")
+        else:
+            self._execute_update_sync_schema(schema, table_name, record, update_action = 'end_prepare')
+
+    def _execute_update_sync_schema(self, schema, table_name, record: [str], update_action: str) -> None:
+        last_successful_sync = record[1]
```

Comment:
"record" is too vague. Unpack the record in the calling code, and maybe even the last_success_ful_sync == None case in that code. Fail early, don't pass empty parameters when not necessary.